packages/changed-elements-react/src/widgets/comparisonJobWidget/hooks/useNamedVersionLoader.tsx

```diff
@@ -175,176 +157,65 @@ const sortAndSetIndexOfNamedVersions = (namedVersions: NamedVersion[], currentNa
-    const loopDelayInMilliseconds = 5000;
-    while (updatingEntries.length > 0 && !args.isDisposed()) {
-      for (let entry of updatingEntries) {
-        await new Promise((resolve) => setTimeout(resolve, loopDelayInMilliseconds));
-        const jobStatusAndJobProgress: JobStatusAndJobProgress = await getJobStatusAndJobProgress(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
-        entry = {
-          version: entry.version,
-          state: VersionProcessedState.Processed,
-          jobStatus: jobStatusAndJobProgress.jobStatus,
-          jobProgress: jobStatusAndJobProgress.jobProgress,
-        };
-        idEntryMap.set(entry.version.id, entry);
-      }
-      entries = [...idEntryMap.values()];
-      updatingEntries = entries.filter((entry) => entry.jobStatus === "Processing" || entry.jobStatus === "Queued");
-      args.namedVersionLoaderState.result = {
-        namedVersions: { currentVersion: args.namedVersionLoaderState.result.namedVersions.currentVersion, entries: entries },
-      };
-      args.setResult(args.namedVersionLoaderState.result);
-    }
-  }
-};
-
-const getJobStatusAndJobProgress = async (comparisonJobClient: IComparisonJobClient, entry: Entry, iTwinId: string, iModelId: string, currentChangesetId: string): Promise<JobStatusAndJobProgress> => {
-  try {
-    const res = await comparisonJobClient.getComparisonJob({
-      iTwinId: iTwinId,
-      iModelId: iModelId,
-      jobId: `${entry.version.changesetId}-${currentChangesetId}`,
-    });
-    if (res) {
-      switch (res.comparisonJob.status) {
-        case "Completed": {
-          return {
-            jobStatus: "Available",
-            jobProgress: {
-              currentProgress: 0,
-              maxProgress: 0,
-            },
-          };
-        }
-        case "Queued": {
-          return {
-            jobStatus: "Queued",
-            jobProgress: {
-              currentProgress: 0,
-              maxProgress: 0,
-            },
-          };
-        }
-        case "Started": {
-          const progressingJob = res as ComparisonJobStarted;
-          return {
-            jobStatus: "Processing",
-            jobProgress: {
-              currentProgress: progressingJob.comparisonJob.currentProgress,
-              maxProgress: progressingJob.comparisonJob.maxProgress,
-            },
-          };
-        }
-        case "Error":
-          return {
-            jobStatus: "Error",
-            jobProgress: {
-              currentProgress: 0,
-              maxProgress: 0,
-            },
-          };
-      }
-    }
-    return {
-      jobStatus: "Unknown",
-      jobProgress: {
-        currentProgress: 0,
-        maxProgress: 0,
-      },
-    };
-  } catch (_) {
-    return {
-      jobStatus: "Not Processed",
-      jobProgress: {
-        currentProgress: 0,
-        maxProgress: 0,
-      },
-    };
-  }
+  args.setResult(args.namedVersionLoaderState);
```

Comment:
This ends up calling a setState function. React does shallow checks for re-renders. I know you are changing the value of the `args.namedVersionLoaderState` up there as a whole, but stylistically, I would prefer if you would simply create an all new variable for the new state instead of changing the input arguments.

Reply:
That's a good suggestion keeps the code clean and make sure the callee does not have any weird side effects